Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Search/OneWay.hs

```diff
@@ -130,6 +133,8 @@ oneWaySearch ::
   m OneWaySearchRes
 oneWaySearch personId req bundleVersion clientVersion device = do
   person <- QP.findById personId >>= fromMaybeM (PersonDoesNotExist personId.getId)
+  mbDisability <- B.runInReplica $ PD.findByPersonId personId
```

Comment:
Just check if `person.hasDisability` is true then only fetch disability else return Nothing
```
mbDisability <- guard person.hasDisability >> B.runInReplica (PD.findByPersonId personId)
```